engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -61,10 +62,15 @@ public class UnionSourceManager {
     private final MergedListener mergedListener;
     private final ConstituentChangesListenerRecorder constituentChangesListener;
     private final UpdateCommitter<UnionSourceManager> updateCommitter;
+    private final ExecutionContext executionContext;
 
     public UnionSourceManager(@NotNull final PartitionedTable partitionedTable) {
         constituentChangesPermitted = partitionedTable.constituentChangesPermitted();
         columnNames = partitionedTable.constituentDefinition().getColumnNamesArray();
+        executionContext = ExecutionContext.newBuilder()
+                .markSystemic()
+                .captureUpdateGraph()
+                .build();
```

Comment:
You should do this after creating the `UpdateCommitter` in the `isRefreshing()` case, and assign it to `null` in the static case.